app/actors/hyrax/actors/doi_actor.rb

```diff
@@ -30,6 +32,10 @@ def create(env)
       #
       # @see Hyrax::Actors::AbstractActor
       def update(env)
+        # Ensure that the work has any changed attributes persisted before we create the job
+        apply_save_data_to_curation_concern(env)
+        save(env)
+
         create_or_update_doi(env.curation_concern) && next_actor.update(env)
```

Comment:
I wonder if the extra save can be avoided by switching the order of operations so the doi job runs last.
```suggestion
        # Ensure that the work has any changed attributes persisted before we create the job
        next_actor.update(env) && create_or_update_doi(env.curation_concern)
```
If this works then the subclassing could probably be reverted to `AbstractActor`.

Reply:
Its a very interesting suggestion, so I tried it out and my test case fails when the order is reversed and the extra save is removed. 

This is the test case: https://github.com/ubiquitypress/hyku_addons/blob/main/spec/features/hyrax_doi_spec.rb

Reply:
Just to make sure I manually tested it and confirmed the test result, that if the extra save isn't there it doesn't seem to create the DOI when updating.

Reply:
That makes sense.  Sad that we need an extra save here, but it probably isn't worth the risk of trying to fix it upstream given that it will be deprecated before too long.

Reply:
It would be nice to get a test for this in `hyrax-doi` but I'm not going to hold up this PR over it.

Reply:
I would have loved to get the test in, but honestly, he time it took to get it working for HA was in the days. So much monkey patching and mocking objects. I did try and ran into road block pretty much straight away.